Fix Astro Island margins in Markdown

HiDeoo opened the pull request:
#### Description

- Closes #3290

This PR attempts to fix an issue where Starlight's Markdown default margins are not applied when using interactive UI frameworks components (Astro Islands).

I don't think we can really reuse our existing rule for this and we may need to have a special case targetting the first child of an island. This PR temporarily adds a few test scenarios to the `/getting-started/` page and I'll also play around with more to see if there are other edge cases we might need to consider.

astrobot-houston commented:
<!-- lunaria-action-comment -->

## Lunaria Status Overview


🌕 **This pull request will trigger status changes.**

<details>
<summary>Learn more</summary>

By default, every PR changing files present in the [Lunaria configuration's `files` property](https://lunaria.dev/reference/configuration/#files-required) will be considered and trigger status changes accordingly.

You can change this by adding one of the keywords present in the [`ignoreKeywords` property in your Lunaria configuration file](https://lunaria.dev/reference/configuration/#ignorekeywords) in the PR's title (ignoring all files) or by [including a tracker directive](https://lunaria.dev/guides/tracking/#tracker-directives) in the merged commit's description. 
</details>

### Tracked Files




| Locale | File                                                                                                                                                          | Note                                                       |
| ------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------- |
| en     | [getting-started.mdx](https://github.com/withastro/starlight/blob/28fd24af382355b7aa4dd9d5e5e02448f501f258/docs%2Fsrc%2Fcontent%2Fdocs%2Fgetting-started.mdx) | Source changed, localizations will be marked as outdated.  |

<details>
<summary>Warnings reference</summary>

| Icon | Description                                                                                                                                       |
| ---- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
| 🔄️  | The source for this localization has been updated since the creation of this pull request, make sure all changes in the source have been applied. |

</details>

delucis reviewed (commented):
Thanks for getting the ball rolling on this one @HiDeoo!

Do you have any more tests that you think we need for confidence? With more realistic scenarios?

In the abstract, this looks OK, but I find it a bit hard to predict what the real-world island usage will actually look like. Especially for slotted content.

HiDeoo commented:
> Do you have any more tests that you think we need for confidence? With more realistic scenarios?

Definitely 👍 That's what I meant in the PR body and I plan on doing that this week-end. I want to try with some random components I use in various projects and test various scenarios, and see if I can find any issues.

Thanks for linking the Cloudflare example, I'll have to test against it too as I mostly worked on relatively simple components so far. I'll undraft the PR once I have done that and feel more confident about the changes.

Regarding the refactor, totally agree, right now I just added new rules and did not even try to see if some smarter refactors could be done or not.

HiDeoo commented:
Added a few more examples, but turns out after checking, I'm not a heavy island user and even less in the middle of content, most of the time it would be in layouts or just Astro components :sweat_smile:

I added one matching the Cloudflare example, a few others that came to mind, but in the end, I still feel like I may not have a full grasp of all or most common use cases of islands in the middle of content.

HiDeoo commented:
As [mentioned](https://github.com/withastro/starlight/pull/3340#issuecomment-3131454150), I've been trying more examples based on searches like https://github.com/search?q=language%3Amdx+%22client%3Aonly%22&type=code&p=1 and a few others. So far, every examples behaved as expected using the new changes from this PR but I'm still not super confident about the change considering I'm not a heavy island user, even more in the middle of content.

I'm opening up this PR for review in case it can bring more awareness to the change and get more feedback from the community if some users are running into similar setups and can to test it out. I will only move forward with the suggested refactoring if we decide to ship this change.

HiDeoo commented:
Updated and cleaned up the PR.

I introduced a new CSS custom property [as suggested](https://github.com/withastro/starlight/pull/3340#pullrequestreview-3055384098) (open on suggestions regarding the naming, e.g. what it represents vs. what it does). I also decided against merging the 2 rules into one, I think it hurts readability too much.

trueberryless commented:
Thanks HiDeoo for the awesome update 🙌
